programs/castle-vault/src/instructions/consolidate_refresh.rs

```diff
@@ -73,20 +82,60 @@ pub fn handler<'info>(ctx: Context<'_, '_, '_, 'info, ConsolidateRefresh<'info>>
 
     // Calculate new vault value
     let vault_reserve_token_amount = ctx.accounts.vault_reserve_token.amount;
-    let vault_value =
-        Provider::iter().try_fold(ctx.accounts.vault_reserve_token.amount, |acc, p| {
-            let allocation = ctx.accounts.vault.actual_allocations[p];
-            if ctx.accounts.vault.get_yield_source_availability(p) {
-                (allocation.last_update.slots_elapsed(clock_slot)? == 0)
-                    .as_result::<u64, ProgramError>(
-                        acc.checked_add(allocation.value)
-                            .ok_or(ErrorCode::OverflowError)?,
-                        ErrorCode::AllocationIsNotUpdated.into(),
-                    )
-            } else {
-                Ok(acc)
+    let vault_value = Provider::iter().try_fold(vault_reserve_token_amount, |acc, p| {
+        let allocation = ctx.accounts.vault.actual_allocations[p];
+        if ctx.accounts.vault.get_yield_source_availability(p) {
+            // We skip pools where we have zero allocation
+            // Must do it inside the if block because we have to check yield source flags first
+            let lp_token_value = match p {
```

Comment:
There's probably a better way to do this with indexing, but should be fine for now--can refactor it later.